linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -221,9 +232,30 @@
     loop_var: definition
   remote_user: "{{ res_def['remote_user'] | default(ansible_user_id) }}"
   delegate_to: "{{ uri_hostname }}"
-  when: res_def['cloud_config'] is defined and node_exists['failed'] is defined and uri_hostname != 'localhost'
+  when: node_exists['failed'] is defined and uri_hostname != 'localhost' and virt_type == "cloud-init"
+
+- name: "Install VM when network_bridge is defined"
+  command: "virt-install --connect {{ definition[6] }} --import --name {{ definition[0] }}_{{ definition[7] }} --autostart --network bridge={{ definition[3] }},model=virtio --ram {{ definition[4] }} --disk path={{ definition[1] }}/{{ definition[0] }}_{{ definition[7] }}.{{ definition[2] }},format={{ definition[5] }},bus=virtio,cache=none --disk path=/tmp/vm-{{ definition[0] }}_{{ definition[7] }}.iso,device=cdrom --wait 10 --os-type=linux --nographics"
+  with_nested:
+    - ["{{ libvirt_resource_name }}"]
+    - ["{{ libvirt_image_path | expanduser }}"]
+    - ["{{ img_src_ext }}"]
+    - ["{{ res_def['network_bridge'] | default('virbr0')  }}"]
+    - ["{{ res_def['memory'] | default(1024)  }}"]
+    - ["{{ res_def['format'] | default('qcow2')  }}"]
+    - ["{{ res_def['uri'] }}"]
+    - "{{ res_count.stdout }}"
+  loop_control:
+    loop_var: definition
+  become: "{{ libvirt_become }}"
+  ignore_errors: yes
+  when: node_exists['failed'] is defined and virt_type == "cloud-init" and cloud_config != {}
 
```

Comment:
This looks like you still have debugging in here and you are running virt-install task twice?  What did the latest run show?  did virt-install run twice or once?

Reply:
Virt-install ran only once since. This task runs when cloud_config is not defined . 
and the following task run when cloud_config is defined

Reply:
I will remove the debug statement .